config/releases.exs

```diff
@@ -32,25 +32,34 @@ defmodule Environment do
     end
   end
 
-  def get_endpoint_static_url_config(nil), do: nil
-  def get_endpoint_static_url_config(""), do: nil
+  def get_endpoint_url_config(nil), do: nil
+  def get_endpoint_url_config(""), do: nil
 
-  def get_endpoint_static_url_config(host) do
+  def get_endpoint_url_config(uri) do
     [
-      host: host,
-      scheme: get("STATIC_URL_SCHEME"),
-      port: get("STATIC_URL_PORT")
+      host: uri.host,
+      scheme: uri.scheme,
+      port: uri.port
     ]
   end
+
+  def get_uri_part(nil, _), do: nil
+  def get_uri_part("", _), do: nil
+  def get_uri_part(uri, :port), do: uri.port
+  def get_uri_part(uri, :scheme), do: uri.scheme
+  def get_uri_part(uri, :host), do: uri.host
+
+  def get_safe_uri(nil), do: nil
+  def get_safe_uri(""), do: nil
+  def get_safe_uri(url), do: URI.parse(url)
 end
 
-force_ssl = Environment.get_boolean("FORCE_SSL")
-scheme = if force_ssl, do: "https", else: "http"
-host = Environment.get("CANONICAL_HOST")
-port = Environment.get("PORT")
+canonical_uri = Environment.get_safe_uri(Environment.get("CANONICAL_URL"))
+static_uri = Environment.get_safe_uri(Environment.get("STATIC_URL"))
+force_ssl = Environment.get_uri_part(canonical_uri, :scheme) == "https"
 
 config :elixir_boilerplate,
-  canonical_host: host,
+  canonical_host: Environment.get_uri_part(canonical_uri, :host),
   force_ssl: force_ssl
```

Comment:
No need for a separate variable since we now use it only once 🎉 

```suggestion
  force_ssl: Environment.get_uri_part(canonical_uri, :scheme) == "https"
```

Reply:
Weird, I missed it, lol! Indeed.